airflow/executors/local_executor.py

```diff
@@ -43,40 +42,44 @@
 This option could lead to the unification of the executor implementations, running
 locally, into just one `LocalExecutor` with multiple modes.
 """
-
-import multiprocessing
 import subprocess
-from queue import Empty
-
-from airflow.executors.base_executor import BaseExecutor
+from multiprocessing import Manager, Process
+from multiprocessing.managers import SyncManager
+from queue import Empty, Queue  # pylint: disable=unused-import  # noqa: F401
+from typing import Any, List, Optional, Tuple, Union  # pylint: disable=unused-import # noqa: F401
+
+from airflow import AirflowException
```

Comment:
Did we not say that inside the core this should be imported from `airflow.exceptions` directly?

Reply:
I don't think it matters that much any more. for sure it does not matter for the "python" cyclic imports (as you explained) but it matters for Pylint ones (those are detected differently - using different algorithms). I think that Pylint cycles are slightly less forgiving but still valid and they are working in a slightly different way (still have not figured all the details). But I think it does not matter any more for most of the imports from airflow (now that we removed operators/hooks/ etc. It still matters for 'models' because we are importing all models there so I prefer to keep 'from airflow.models.baseoperator import BaseOperator' in the core rather than 'from airflow.models import BaseOperator'. We might revisit that in the future however.